applications/nrf5340_audio/src/modules/sd_card.h

```diff
@@ -52,6 +54,51 @@ int sd_card_write(char const *const filename, char const *const data, size_t *si
  */
 int sd_card_read(char const *const filename, char *const data, size_t *size);
 
+/**@brief   Open file on SD card
+ * param[in] filename   Name of file to open
+ * param[in] path_to_file Path to file
+ * @retval  0 on success
+ *              -ENODEV SD init failed. SD likely not inserted
+ *      Otherwise, error from underlying drivers
+ */
+int sd_card_segment_open(char const *const filename, char const *const path_to_file);
+
+/**@brief   Read segment on the open file on the SD card
+ * param[in] data   Where the read data is stored
+ * @param[in,out] size	Number of bytes to be read from file
+ *			The actual read size will be returned
+ *			If the actual read size is 0, there will be a warning message which
+ *			indicates that the file is empty
+ * @retval  0 on success
+ *              -ENODEV SD init failed. SD likely not inserted
+ *      Otherwise, error from underlying drivers
+ */
+int sd_card_segment_read(char *const data, size_t *size);
+
+/**@brief   Close the file opened by the sd_card_segment_read_open function
+ * @retval  0 on success
+ *              -EBUSY Segment read operation has not started
+ *      Otherwise, error from underlying drivers
+ */
+int sd_card_segment_close(void);
+
+/**@brief   Peek at data ahead of current file position without changing file position
+ * param[in, out] data Location to where the read data should be stored
+ * param[in, out] size Size of data to be read
```

Comment:
```suggestion
 * @param[in, out] data Location to where the read data should be stored
 * @param[in, out] size Size of data to be read
```

Reply:
How is data an out? Is the pointer changed? You can probably just remove all the [in] and [in, out] from this file to keep it simpler

Reply:
I still suggest removing all the [in] and [in, out] from this file.